packages/checkbox/src/useCheckbox.ts

```diff
@@ -1,23 +1,183 @@
-import { ref, computed, inject } from 'vue'
+import {
+  ref,
+  computed,
+  inject,
+  getCurrentInstance,
+  onMounted,
+  watch,
+} from 'vue'
+import { toTypeString } from '@vue/shared'
+import { UPDATE_MODEL_EVENT } from '@element-plus/utils/constants'
+import { ICheckboxGroupInstance, ICheckboxProps, PartialReturnType } from './checkbox'
 
-export const useCheckbox = () => {
+export const useCheckboxGroup = () => {
   //todo: ELEMENT
   const ELEMENT = null
   const elForm = inject('elForm', {})
   const elFormItem = inject('elFormItem', {}) as any
-  const _checkboxGroup = inject('CheckboxGroup', {}) as any
-  const focus = ref(false)
-  const isGroup = computed(() => _checkboxGroup && _checkboxGroup.name === 'ElCheckboxGroup')
-  const _elFormItemSize = computed(() => {
+  const checkboxGroup = inject<ICheckboxGroupInstance>('CheckboxGroup', {})
+  const isGroup = computed(() => checkboxGroup && checkboxGroup.name === 'ElCheckboxGroup')
+  const elFormItemSize = computed(() => {
     return (elFormItem || {} as any).elFormItemSize
   })
   return {
     isGroup,
-    focus,
-    _checkboxGroup,
+    checkboxGroup,
     elForm,
     ELEMENT,
-    _elFormItemSize,
+    elFormItemSize,
     elFormItem,
   }
 }
+
+const useModel = (props: ICheckboxProps) => {
+  let selfModel = false
+  const { emit } = getCurrentInstance()
+  const { isGroup, checkboxGroup } = useCheckboxGroup()
+  const isLimitExceeded = ref(false)
+  const store = computed(() => checkboxGroup ? checkboxGroup.modelValue?.value : props.modelValue)
+  const model = computed({
+    get() {
+      return isGroup.value
+        ? store.value
+        : props.modelValue ?? selfModel
+    },
```

Comment:
This is good, because I have seen many components still tries to implement compatible API for both `value & v-model`
Can we make this to a general compositions? Like taking a `function` as parameter to handling user defined stuffs

Reply:
Yep, maybe we need more discuss